gps_error_simulator:
  ros__parameters:
    use_sim_time: True
    input_topic: '/emlid/fix'
    output_topic: '/emlid/modified'
    enable_logs: True
    enable_mysterious_offset_correction: True
    enable_noise: True
    enable_offset: False
    enable_spikes: False
    enable_delay: False
    enable_covariance_set: False        # todo remove this and just use default_covariance, which is usually 0.0. 
    enable_covariance_increase: False   # todo maybe remove. If true, covariance_increase_. will be added to default covariance. Overwritten by truthful covariance
    enable_truthful_covariance: False   # Only effective if noise, spikes or offset enabled. Sets covariance corresponding to error. Overwrites other covariance modifiers.
    noise_sigma: 0.1                 # Gauss sigma is set in m. 99.7% of errors less than 10cm => sigma = 1/3 * 0.1 is the same as 95% less than 6.6cm => sigma = 1/2 * 0.066
    restrain_noise_to_x_direction: False
    restrain_noise_to_y_direction: False # If x and y direction set to true, noise is restrained to x direction
    offset_x: 0.25                      # in meter, only if enable_offset is set
    offset_y: 0.25
    delay: 2.0                          # in seconds , only if enable_delay is set
    spikes_probability: 0.0
    spikes_mean: 0.0
    spikes_std: 0.0
    #spikes_direction: 0.0              # If not set, random direction. Set in radians to limit spikes to one direction. 0.0 is x direction (east)
    covariance_set_x: 0.0               # Default (if enable_covariance_set), base for increase
    covariance_set_y: 0.0
    covariance_increase_x: 0.0
    covariance_increase_y: 0.0
